Added Maestro Cloud Upload component doc (#421)
appcircleio/appcircle-docusaurus: docs/workflows/common-workflow-steps/maestro-cloud-upload.md

---
title: Maestro Cloud Upload
description: Facilitate the execution of automated tests for mobile applications directly within Appcircle using Maestro Cloud Upload.
tags: [maestro, maestro-cloud, mobile, testing]
sidebar_position: 11
---

import Screenshot from '@site/src/components/Screenshot';

# Maestro Cloud Upload

> [Maestro Cloud](https://cloud.mobile.dev) is a cloud-based mobile device farm designed for testing and debugging mobile applications. It provides developers and QA teams with access to a wide range of real devices, allowing them to perform thorough testing across different platforms, operating systems, and device configurations. 

Reviewer: ```suggestion
[Maestro Cloud](https://cloud.mobile.dev) is a cloud-based mobile device farm for testing and debugging mobile applications, offering developers and QA teams access to a diverse array of real devices for comprehensive testing across various platforms, operating systems, and device configurations.
```

| `$AC_MAESTRO_EXPORT_OUTPUT`   | Specify the output file for the test file output. The default is `report.xml`.                              | Optional  |
| `$AC_MAESTRO_MAPPING_FILE`    | **Android**: Include the Proguard mapping file. **iOS**: Include the generated .dSYM file.              | Optional  |
| `$AC_MAESTRO_BRANCH`          | The branch from which the upload originated.                                                            | Optional  |
| `$AC_MAESTRO_REPO_NAME`       | The name of the repository (e.g., GitHub repo slug).                                                     | Optional  |
| `$AC_MAESTRO_REPO_OWNER`      | The owner of the repository (e.g., GitHub organization or user slug).                                    | Optional  |
| `$AC_MAESTRO_PULL_ID`         | The ID of the pull request from which the upload originated.                                            | Optional  |
| `$AC_MAESTRO_CLI_VERSION`     | The version of the Maestro CLI is to be downloaded in your CI environment. The default value is the `latest` version.  | Optional  |

### Output Variables

> The **Maestro Cloud Upload** step does not produce any output variables. The results are shown in the build log.

Reviewer: ```suggestion
No output variables are produced by the **Maestro Cloud Upload** step; results are displayed in the build log.
```

Author: @erenkan I am in favor of keeping the old sentence unchanged for the following reasons:
1. Active Language: The second sentence ("The Maestro Cloud Upload step does not produce any output variables.") uses active language, making it more engaging and direct compared to the passive construction in the first sentence ("No output variables are produced by the Maestro Cloud Upload step").
2. Consistency: The second sentence maintains consistency by using "The Maestro Cloud Upload step" as the subject, which matches the style used in the rest of the document. Consistency in style enhances readability and clarity for the reader.
